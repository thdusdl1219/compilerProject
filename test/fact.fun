fun factorial(n : int):int = 
  if (n = 0) then 1
  else if (n < 0) then 1
  else n * factorial(n - 1)

fun main(argc:int):int = 
  printint(factorial(10)); 0
