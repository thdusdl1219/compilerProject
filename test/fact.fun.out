[3628800, 0]
